x-pack/plugins/file_upload/public/api/index.ts

```diff
@@ -30,3 +32,23 @@ export async function importerFactory(
   const fileUploadModules = await lazyLoadFileUploadModules();
   return fileUploadModules.importerFactory(format, options);
 }
+
+interface HasImportPermissionParams {
+  checkCreateIndexPattern: boolean;
+  hasPipeline: boolean;
+  indexName?: string;
+}
+
+export async function hasImportPermission(params: HasImportPermissionParams): Promise<boolean> {
+  const fileUploadModules = await lazyLoadFileUploadModules();
+  try {
+    const resp = await fileUploadModules.getHttp().fetch<HasImportPermission>({
+      path: `/internal/file_upload/has_import_permission`,
+      method: 'GET',
+      query: { ...params },
```

Comment:
Can we just pass `params` without the spread operator inside an object?

Reply:
For some reason just using params gives some typescript errors

```
x-pack/plugins/file_upload/public/api/index.ts:45:24 - error TS2769: No overload matches this call.
  Overload 1 of 4, '(options: HttpFetchOptionsWithPath & { asResponse: true; }): Promise<HttpResponse<HasImportPermission>>', gave the following error.
    Type 'HasImportPermissionParams' is not assignable to type 'HttpFetchQuery'.
      Index signature is missing in type 'HasImportPermissionParams'.
  Overload 2 of 4, '(path: string, options?: HttpFetchOptions | undefined): Promise<HasImportPermission>', gave the following error.
    Argument of type '{ path: string; method: string; query: HasImportPermissionParams; }' is not assignable to parameter of type 'string'.
  Overload 3 of 4, '(options: HttpFetchOptionsWithPath): Promise<HasImportPermission>', gave the following error.
    Type 'HasImportPermissionParams' is not assignable to type 'HttpFetchQuery'.

 45     const resp = await fileUploadModules.getHttp().fetch<HasImportPermission>({
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 46       path: `/internal/file_upload/has_import_permission`,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 48       query: params,
    ~~~~~~~~~~~~~~~~~~~~
 49     });
    ~~~~~~

  src/core/target/types/public/http/types.d.ts:210:5
    210     query?: HttpFetchQuery;
            ~~~~~
    The expected type comes from property 'query' which is declared here on type 'HttpFetchOptionsWithPath & { asResponse: true; }'
  src/core/target/types/public/http/types.d.ts:210:5
    210     query?: HttpFetchQuery;
            ~~~~~
    The expected type comes from property 'query' which is declared here on type 'HttpFetchOptionsWithPath'

x-pack/plugins/file_upload/public/api/index.ts:50:17 - error TS2339: Property 'hasImportPermission' does not exist on type 'HttpResponse<HasImportPermission>'.

50     return resp.hasImportPermission;
```